shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -229,9 +225,18 @@ - (BOOL)application:(UIApplication*)application
     continueUserActivity:(NSUserActivity*)userActivity
       restorationHandler:(void (^)(NSArray* __nullable restorableObjects))restorationHandler {
 #endif
-  return [_lifeCycleDelegate application:application
-                    continueUserActivity:userActivity
-                      restorationHandler:restorationHandler];
+  if ([_lifeCycleDelegate application:application
+                 continueUserActivity:userActivity
+                   restorationHandler:restorationHandler]) {
+    return YES;
+  }
+  if (userActivity.activityType == NSUserActivityTypeBrowsingWeb) {
+    return NO;
+  }
```

Comment:
I'm not catching why this is here.  It sounds like `NSUserActivityTypeBrowsingWeb` is associated with universal links?  Wouldn't you want this?  I see this behavior matches what you want in your test so I assume it's true, I'm just not familiar with the API.

Maybe a few words as to why we return NO here?

Reply:
Actually, this is a good question.  @AlBlanc I noticed you did you did this in https://github.com/flutter/engine/pull/25705 and meant to follow up.  Why did you avoid this activity type?

Reply:
I don't see a reason to not process `NSUserActivityTypeBrowsingWeb` so I removed this and the test.
https://developer.apple.com/documentation/xcode/supporting-universal-links-in-your-app

We can always add it back later if there's some good reason to.

Reply:
That sounds fine with me, we could wait a bit more to give @alblanc 24 hours to respond.  He's probably celebrating the France basketball teams win =)

Reply:
I was thinking I can merge this now and put the additional PR if should be excluded.  Based on my reading of https://developer.apple.com/documentation/xcode/supporting-universal-links-in-your-app it shouldn't be.
But yeah I can wait too, it's not a rush...

Reply:
I'm going to merge this. I'll change it if we hear we need to exclude `NSUserActivityTypeBrowsingWeb` for some reason.